### PR TITLE
ISPN-6074 DistExec API should not require a distributed cache

### PR DESCRIPTION
**Pull request by wburns**
* Added ClusterExecutor interface

https://issues.jboss.org/browse/ISPN-6074

### CODE REVIEW

File: core/src/main/java/org/infinispan/manager/DefaultCacheManager.java [hunk 1]
```diff
@@ -936,4 +942,19 @@ public Properties getGlobalConfigurationAsProperties() {
    public CacheContainerStats getStats() {
       return stats;
    }
+
+   @Override
+   public ClusterExecutor executor() {
+      if (!started) {
```
Comment (tristantarrant) [inline core/src/main/java/org/infinispan/manager/DefaultCacheManager.java]: You could do away with the "started" field and check on the status of the GCR

File: core/src/main/java/org/infinispan/manager/impl/ClusterExecutorImpl.java [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+                 mode, unit.toMillis(time), DeliverOrder.NONE).handle((r, t) -> {
+            if (t != null) {
+               rethrowException(t);
+            }
+            if (r.wasReceived()) {
+               if (r.hasException()) {
+                  rethrowException(r.getException());
+               }
+            } else if (r.wasSuspected()) {
+               throw new SuspectException("Node " + JGroupsAddressCache.fromJGroupsAddress(target) + " was suspected");
```
Comment (tristantarrant) [inline core/src/main/java/org/infinispan/manager/impl/ClusterExecutorImpl.java]: I think these should be i18n'ed

File: core/src/main/java/org/infinispan/manager/DefaultCacheManager.java (hunk 1 above)
Comment (wburns) [inline core/src/main/java/org/infinispan/manager/DefaultCacheManager.java]: Makes sense, thanks.

**Comment by wburns** [comment]
This should be good now after https://github.com/infinispan/infinispan/pull/4011 is integrated

**Comment by galderz** [comment]
Need rebase

File: core/src/main/java/org/infinispan/manager/impl/ClusterExecutorImpl.java [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+      } else {
+         // If there was no transport this cache manager cannot have a cluster, so we just always execute
+         this.me = null;
+      }
+      this.time = time;
+      this.unit = unit;
+      this.localExecutor = localExecutor;
+   }
+
+   static final class RunnableAdapter<T> implements Serializable, Callable<T> {
```
Comment (galderz) [inline core/src/main/java/org/infinispan/manager/impl/ClusterExecutorImpl.java]: Never used

Comment (wburns) [inline core/src/main/java/org/infinispan/manager/impl/ClusterExecutorImpl.java]: Ah yes sorry was used in a previous iteration :+1:

Comment (wburns) [inline core/src/main/java/org/infinispan/manager/impl/ClusterExecutorImpl.java]: Removed

**Comment by wburns** [comment]
Updated

**Comment by galderz** [comment]
Looks good, testing... :)

**Comment by galderz** [comment]
Integrated, thanks @wburns :)

**Pull request closed without merging**